libr/fs/fs.c

```diff
@@ -700,13 +702,17 @@ R_API int r_fs_prompt(RFS* fs, const char* root) {
 	}
 
 	for (;;) {
-		printf ("[%s]> ", path);
-		fflush (stdout);
-		fgets (buf, sizeof (buf) - 1, stdin);
-		if (feof (stdin)) {
+		strncpy (prompt, path, sizeof (prompt) - 1);
+		strcat (prompt, "> ");
```

Comment:
This will overflow if the previous strncpy fills the buffer. And its doing an unnecessary strlen on the string copied in the line above. Just do snprintf